### DISCUSSION TITLE
Is there any reason why Meson doesn't support Makefiles?

### DISCUSSION THREAD

**Discussion by olivia-banks** [mesonbuild/meson#10125] [Q&A] [answered]
Is there any reason why Meson doesn't support Makefiles (e.g. it goes against a strict design philosophy)? If there's not, and no one's gotten around to it, I'd like to submit a pull request implementing a Makefile backend.

Really don't want to have to switch to CMake just for Makefile support!

Sorry, no idea why there is a duplicate discussion.

**Comment by xclaesse** [comment, marked as answer]
Let's reply with another question: is there any reason you need makefile and can't use Ninja?

**Reply by olivia-banks** [reply]
Project requirements. I'd gladly take Ninja any day otherwise, however. Who can pass up fast build times like that :)

Also, this discussion was converted from an issue. See [here](https://github.com/mesonbuild/meson/discussions/10124#discussion-3937827).

**Reply by xclaesse** [reply]
"Your requirements are definitely dumb, it does not matter who gave them to you." - Elon Musk. Not a fan of that guy, but that's one thing I totally agree with him, people does not question requirements enough.

**Reply by olivia-banks** [reply]
I questioned them and got a one-word answer, and that was "compatibility". With what exactly is beyond me.

**Reply by eli-schwartz** [reply]
Apparently some machine out there that:
- supports python 3.7
- supports make
- supports git
- supports downloading or copying your project

does NOT:
- support downloading or copying https://github.com/michaelforney/samurai

It's not Windows, macOS, or Linux, because all 3 have first class ninja support and binary releases.

It's not POSIX, because samurai isn't supported.

It's not OpenVMS, that needs MMS and not Make.

Hmm, the mystery is fascinating.

**Reply by olivia-banks** [reply]
I was going to say Plan9, but that's POSIX and offers the utilities to use Meson (with some work). Perhaps it's an old version of L4?

**Reply by jpakkane** [reply]
Or something like IBM zOS?

**Reply by dcbaker** [reply]
The only case I’ve run into, is at least one of the BSDs have a requirement that the core OS must be built with their make implementation. But they don’t care at all about ports, so unless clang switches to Meson I don’t think that would be a problem.

**Reply by eli-schwartz** [reply]
Clang doesn't build with make unless you also run cmake.

I would expect that they ported clang to not need a configure stage since "we know exactly what we need for BSD core", knowing them.

**Reply by dcbaker** [reply]
I think they run cmake to generate make files, then check those in, actually... At least, that's what they do with autoconf

**Reply by eli-schwartz** [reply]
How would that even work, since autotools is designed to create standalone Makefiles (plus rules to regenerate them using autotools) whereas cmake creates Makefiles that call back into the cmake binary for the same reason meson has `meson --internal` (except cmake is ***much*** happier to use the cmake program for every goshdarned little thing).
